src/battle-dex.ts

```diff
@@ -173,10 +173,16 @@ const Dex = new class implements ModdedDex {
 	readonly gen = 8;
 	readonly modid = 'gen8' as ID;
 	readonly cache = null!;
-
+	readonly modCache = {
+		Moves: {} as any as {[k: string]: Move},
+		Items: {} as any as {[k: string]: Item},
+		Abilities: {} as any as {[k: string]: Ability},
+		Species: {} as any as {[k: string]: Species},
+		Types: {} as any as {[k: string]: Effect},
+	};
 	readonly statNames: ReadonlyArray<StatName> = ['hp', 'atk', 'def', 'spa', 'spd', 'spe'];
 	readonly statNamesExceptHP: ReadonlyArray<StatNameExceptHP> = ['atk', 'def', 'spa', 'spd', 'spe'];
-
+	modData?: any;
```

Comment:
`modData` being type `any` seems odd, it'd be preferable to have it be typed something like:
```js
{
  Moves?: {[k: string]: {[k: string]: any}},
  Abilities?: {[k: string]: {[k: string]: any}},
  ...etc
}
```
so there's a slight idea of what's in the object